launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -614,12 +618,39 @@ QProcessEnvironment MinecraftInstance::createLaunchEnvironment()
     }
 
     if (settings()->get("UseDiscreteGpu").toBool()) {
+#ifdef WITH_QTDBUS
+        if (QDBusConnection::systemBus().isConnected()) {
+            QDBusInterface switcheroo("net.hadess.SwitcherooControl", "/net/hadess/SwitcherooControl", "org.freedesktop.DBus.Properties", QDBusConnection::systemBus());
+
+            if (switcheroo.isValid()) {
+                QDBusReply<QDBusVariant> reply = switcheroo.call(QStringLiteral("Get"), QStringLiteral("net.hadess.SwitcherooControl"), QStringLiteral("GPUs"));
+                if (reply.isValid()) {
+                    QDBusArgument arg = qvariant_cast<QDBusArgument>(reply.value().variant());
+                    QList<QVariantMap> gpus;
+                    arg >> gpus;
+
+                    for (const auto &gpu : gpus) {
+                        QString name = qvariant_cast<QString>(gpu[QStringLiteral("Name")]);
+                        bool defaultGpu = qvariant_cast<bool>(gpu[QStringLiteral("Default")]);
+                        if (!defaultGpu) {
+                            QStringList envList = qvariant_cast<QStringList>(gpu[QStringLiteral("Environment")]);                            
+                            for (int i = 0; i + 1 < envList.size(); i += 2) {
+                                env.insert(envList[i], envList[i + 1]);
+                            }
+                            break;
+                        }
+                    }
+                }
```

Comment:
Ok what happens in case it is built with Dbus support but SwitcherooControl is not installed?

Reply:
With the implementation as it is right now it would not do anything if switcheroo-control is not available.
Overall my original implementation (the old code) is very naive and causes more problems than it helps to fix so I think it should be gutted as soon as possible
I'm fine to adapt the code to try using switcheroo-control and fallback to the old behavior and otherwise just use the old behavior if qtdbus is not available
after this PR I plan on splitting the discrete GPU code into a dedicated class and turning the simple checkbox to enable discrete GPU support into a combo-box where you can choose between some specfic GPU or the default.

Reply:
Please also add a fallback in case the switcheroo-control is not available
With this build the integrated GPU is always selected for me (I do not have SwitcherooControl but it was built wit qtBus support)